spec/coffee/bootstrap.coffee

```diff
@@ -1,6 +1,48 @@
 #global jasmine protects
 beforeEach ->
   @googleTemp = window.google
+
+  angular.module('uiGmapgoogle-maps')
+  .config ($provide) ->
+    $provide.decorator '$timeout', ($delegate, $browser) ->
+      $delegate.hasPendingTasks = ->
+        $browser.deferredFns.length > 0
+
+      $delegate
+
+  #create
+  angular.module('uiGmapgoogle-maps.specs', ['uiGmapgoogle-maps'])
+  #require for testing
+  module "uiGmapgoogle-maps.specs"
+
+  @injectAll = =>
+    @injects.forEach (toInject) ->
+      inject toInject
+
+  @injects = []
+
+  @injects.push (_$rootScope_, $timeout, $q, $browser, $compile) =>
+    @q = $q
+    @browser = $browser
+    @rootScope = _$rootScope_
+    @scope = angular.extend @rootScope.$new(), @scope or {}
+    @scope.map = angular.extend @scope.map or {},
+      zoom: 12
+      center:
+        longitude: 47
+        latitude: -27
+        
+    @timeout = $timeout
+    @compile = $compile
+
+  @digest = (fn) =>
+    @compile(@html)(@scope) if @html and @scope
```

Comment:
compile portions might not be desired; but it does eliminate many lines in directive instantiation.